src/pydata_sphinx_theme/theme/pydata_sphinx_theme/components/search-field.html

```diff
@@ -2,13 +2,13 @@
 <form class="bd-search d-flex align-items-center"
       action="{{ pathto('search') }}"
       method="get">
-  <i class="fa-solid fa-magnifying-glass"></i>
+  <label for="{{ search_input_id }}"
+         aria-label="{{ theme_search_bar_text }}"><i class="fa-solid fa-magnifying-glass"></i></label>
```

Comment:
Side note: the `placeholder` value on the input is surfaced to VoiceOver on Mac. I wonder if that's true for other screen readers. If so, we should probably just mark this label with aria-hidden.